File: 2325-decode-the-message/2325-decode-the-message.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        d = {}
        ans = ""
        value = 0
        
        for i in key:
            if i != " ":
                if d.get(i) == None:
                    d[i] = 97 + value
                    value = value + 1
        
        for i in message:
            if i == " ":
                ans += " "
            else:
                ans += chr(d.get(i))
        
        return ans

File: 2325-decode-the-message/2325-decode-the-message.cpp
int hsh[26];

class Solution {
public:
    string decodeMessage(string key, string message) {
        for (int i = 0; i < 26; ++i) hsh[i] = -1;
        int i = 0;
        string ans;
        for (auto &ele : key) {
            if (ele != ' ')
                if (hsh[ele - 'a'] == -1)
                    hsh[ele - 'a'] = i++;
        }
        for (auto &ele : message) {
            if (ele != ' ')
                ans.push_back(97 + hsh[ele - 'a']);
            else
                ans.push_back(' ');
        }
        return ans;
    }
};